### PR TITLE
[MNG-7038] Fixes for topDirectory

### PR DESCRIPTION
**Pull request by cstamas**
Changes:
* resolve file argument against workingDirectory explicitly
* the check Files.isRegularFile should happen against path

### CODE REVIEW

File: maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java [hunk 1]
```diff
@@ -332,10 +332,10 @@ void initialize(CliRequest cliRequest) throws ExitException {
         for (String arg : cliRequest.args) {
             if (isAltFile) {
                 // this is the argument following -f/--file
-                Path path = Paths.get(arg);
+                Path path = topDirectory.resolve(arg);
```
Comment (gnodet) [inline maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java:335]: That's really weird.  That would make file argument order dependent as the `topDirectory` is modified inside the loop.

Comment (gnodet) [reply maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java:335]: Even if this was not the case, the goal of the loop is to compute the `topDirectory`.
However, introducing a `workingDirectory` variable and using it to resolve the `path` could make sense.

Comment (cstamas) [reply maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java:335]: I still don't understand why would this loop enter 2 or more times? (the user specifies several -f parameters?)

Comment (cstamas) [reply maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java:335]: but even then, there is a break (or throws) so first entrance will either break or throw

Comment (gnodet) [reply maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java:335]: Missed the `break` ...

**Review by gnodet — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #8000

**Pull request merged**